shared/js/project.js

```diff
@@ -464,6 +496,13 @@ function executeCallback(callback, data) {
 }
 
 /**
+ * is the current project (if any) editable by the logged in user (if any)?
+ */
+function isEditable() {
```

Comment:
hm... I think I'm going to punt on this for reasons that I think I'm going to deal with this again in the submit flow which is my next work item (behavior for this feature: submitting a solution to a level makes it uneditable; this applies to levels in a script, not to standalone projects). This might be the same state as this frozen (there are some other subtleties like the teacher is able to unfreeze it, and this state also needs to be reflected in the level progress and implemented for non-project backed levels).